packages/sit-onyx/src/components/OnyxListbox/OnyxListbox.vue

```diff
@@ -52,85 +58,133 @@ const slots = defineSlots<{
 
 const { t } = injectI18n();
 
+const isExpanded = ref(false);
+
+watchEffect(async () => {
+  if (isExpanded.value) {
+    await nextTick();
+    miniSearch.value?.focus();
+  }
+});
+
 /**
  * Currently (visually) active option.
  */
-const activeOption = ref<TValue>();
+const activeValue = ref<TValue>();
 
-const searchInput = computed({
-  get: () => (props.withSearch && props.searchTerm) || "",
-  set: (newValue) => emit("update:searchTerm", newValue),
+const arrayValue = computed(() => {
+  if (!props.modelValue) {
+    return [];
+  }
+  return props.multiple ? props.modelValue : [props.modelValue];
 });
 
+const miniSearch = ref<InstanceType<typeof OnyxMiniSearch>>();
+const selectInput = ref<ComponentExposed<typeof OnyxSelectInput>>();
+
 /**
  * Sync the active option with the selected option on single select.
  */
-watch(
-  () => props.modelValue,
-  (newValue) => {
-    if (!props.multiple) {
-      activeOption.value = newValue as typeof activeOption.value;
-    }
-  },
-);
+watch(arrayValue, () => {
+  if (!props.multiple) {
+    activeValue.value = arrayValue.value.at(0)?.value;
+  }
+});
 
 /** unique ID to identify the `select all` checkbox */
-const CHECK_ALL_ID = createId("ONYX_CHECK_ALL") as TValue;
+const CHECK_ALL_ID = createId("ONYX_CHECK_ALL");
 
 /**
  * IDs of all options that can be navigated with the keyboard.
  * Includes "select all" up front if it is used.
  */
 const allKeyboardOptionIds = computed(() => {
-  return (props.multiple && props.withCheckAll ? [CHECK_ALL_ID] : []).concat(
+  return (props.multiple && props.withCheckAll ? [CHECK_ALL_ID as TValue] : []).concat(
     enabledOptionValues.value,
   );
 });
 
-const {
-  elements: { listbox, option: headlessOption, group: headlessGroup },
-} = createListbox({
-  label: computed(() => props.label),
-  multiple: computed(() => !!props.multiple),
-  activeOption,
-  onSelect: (selectedOption) => {
-    if (selectedOption === CHECK_ALL_ID) {
-      checkAll.value?.handleChange(!checkAll.value.state.value.modelValue);
-      return;
-    }
+const onToggle = () => {
+  isExpanded.value = !isExpanded.value;
+  if (!isExpanded.value) {
```

Comment:
We could also move the watchEffect above into here
![image](https://github.com/SchwarzIT/onyx/assets/67898185/75dbcb8e-d3e9-4619-826e-1c8cfe588e1a)